app/models/tenant.rb

```diff
@@ -292,6 +292,8 @@ def allowed?
   end
 
   def create_miq_product_features_for_tenant_nodes
+    return [] if !miq_region || MiqRegion.my_region_number != miq_region.region
+
     result = MiqProductFeature.with_tenant_feature_root_features.map.each do |tenant_miq_product_feature|
```

Comment:
@lpichler I might not be understanding the issue fully but wouldn't the fix be to scope this query to the current region like this?

`result = MiqProductFeature.in_my_region.with_tenant_feature_root_features.map.each do |tenant_miq_product_feature|`

Then it would never involve product features from other regions